File: trunk/resources/scop_1.5.1/python/examples/xml_sender.py
#!/usr/bin/python
# xml_sender.py - DMI - 11-11-03

import scop

def marshall(dict):
	l = []
	keys = dict.keys()
	for k in keys:
		l.append([k, dict[k]])
	return l

ab = { "Poirot": "Belgium",
       "Morse": "Oxford, UK",
       "Danger Mouse": "London, UK" }
v = marshall(ab)
sock = scop.scop_open("localhost", "xml_sender")
scop.scop_send_struct(sock, "xml_receiver", v)
scop.scop_close(sock)

File: trunk/resources/scop_1.5.1/examples/xml_sender.cpp
// xml_sender.cpp - DMI - 7-9-02

#include <scop.h>
#include <scopxml.h>

#include "address_book.h"

int main()
{
	int sock;
	AddressBook ab(3);
	vertex *v;
	
	ab.set_entry(0, "Poirot", "Belgium");
	ab.set_entry(1, "Morse", "Oxford, UK");
	ab.set_entry(2, "Danger Mouse", "London, UK");
	v = ab.marshall();
	
	sock = scop_open("localhost", "xml_sender");
	scop_send_struct(sock, "xml_receiver", v);
	delete v;
	
	close(sock);
	return 0;
}
